WordPress/src/main/java/org/wordpress/android/ui/posts/adapters/AuthorSelectionAdapter.kt

```diff
@@ -0,0 +1,171 @@
+
+    override fun getDropDownView(position: Int, convertView: View?, parent: ViewGroup): View {
+        var view: View? = convertView
+        val holder: DropdownViewHolder
+
+        if (view == null) {
+            val inflater = LayoutInflater.from(parent.context)
+            view = inflater.inflate(R.layout.post_list_author_selection_dropdown, parent, false)
+            holder = DropdownViewHolder(view)
+            view.tag = holder
+        } else {
+            holder = view.tag as DropdownViewHolder
+        }
+
+        holder.bind(items[position], imageManager, uiHelpers)
+
+        return view!!
+    }
+
+    override fun getItem(position: Int): AuthorFilterListItemUIState = items[position]
+
+    override fun getItemId(position: Int): Long = position.toLong()
+
+    override fun getItemViewType(position: Int): Int = 0
+
+    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
+        var view: View? = convertView
+        val holder: NormalViewHolder
+
+        if (view == null) {
+            val inflater = LayoutInflater.from(parent.context)
+            view = inflater.inflate(R.layout.post_list_author_selection, parent, false)
+            holder = NormalViewHolder(view)
+            view.tag = holder
+        } else {
+            holder = view.tag as NormalViewHolder
+        }
+
+        holder.bind(items[position], imageManager, uiHelpers)
+
+        return view!!
+    }
+
+    override fun getViewTypeCount(): Int = 1
+    override fun hasStableIds(): Boolean = false
+    override fun isEmpty(): Boolean = false
+
+    fun notifyDataSetChanged() {
+        for (observer in observers) {
+            observer.get()?.onChanged()
+        }
+    }
+
+    override fun registerDataSetObserver(observer: DataSetObserver) {
+        for (alreadyAdded in observers) {
+            if (observer == alreadyAdded.get()) {
+                // No need to add since it is already added
+                return
+            }
+        }
+
+        observers.add(WeakReference(observer))
+    }
+
+    fun updateItems(newItems: List<AuthorFilterListItemUIState>) {
+        items.clear()
+        items.addAll(newItems)
+        notifyDataSetChanged()
+    }
+
+    override fun unregisterDataSetObserver(observer: DataSetObserver) {
+        for ((index, currentRef) in observers.withIndex()) {
+            if (observer == currentRef.get()) {
+                observers.removeAt(index)
+                return
+            }
+        }
+    }
+
+    open class NormalViewHolder(protected val itemView: View) {
+        protected val image: AppCompatImageView = itemView.findViewById(R.id.post_list_author_selection_image)
+
+        @CallSuper
+        open fun bind(state: AuthorFilterListItemUIState, imageManager: ImageManager, uiHelpers: UiHelpers) {
+            val context = itemView.context
+            val avatarUrl: String? = (state as? AuthorFilterListItemUIState.Me)?.avatarUrl
+
+            val padding: Int
```

Comment:
If you decide to go in this path, you may also consider creating a sealed class so we can avoid inheritance... something like

```
sealed class AuthorSelectionViewHolder(..) {
   fun bind(...) {
       imageManager.loadIntoCircle(...)
   }
   class NormalViewHolder(...): AuthorSelectionViewHolder(..)
   class DropdownViewHolder(..) : AuthorSelectionViewHolder(..)
```

 Wdyt?

Reply:
I think my only concern with this is that the placeholder/fallback icon for AVATAR_WITHOUT_BACKGROUND is a different colour to what we would use for MULTI_USER_AVATAR_WHITE_BACKGROUND.

Which I can see resolving one of 2 ways:
1. Make an AVATAR_WITHOUT_BACKGROUND variant for the correct colour (the benefit of which is that we could have use `ic_user_white_24dp` instead of `ic_user_circle_grey_24dp` for consistency with `ic_multiple_users_white_24dp`)
1. Change our the multiple user drawable to be the same colour as avatar is now.

I'm hoping the specific colour wasn't vital to the design and we can go with option 2

Reply:
One of the reasons I decided not to allow customization in the `ImagePlaceholderManager` was to "enforce" consistency across the app. On the other hand I'm a bit worried the 'grey' background won't look so good on the blue toolbar.

@SylvesterWilmott Could we invert the colors for the MULTIUSER selection filter icon -> grey background white icon?

![multiuser_filter](https://user-images.githubusercontent.com/2261188/54863935-3f398480-4d4f-11e9-805e-7bcc03f21571.jpg)


Note:  If we decide to introduce `MULTI_USER_AVATAR_WHITE_BACKGROUND` we should ideally rename the `AVATAR_WITH_BACKGROUND` to `AVATAR_WITH_GREY_BACKGROUND` but I wouldn't do it in this PR.

Reply:
@SylvesterWilmott Let me clarify. We currently use these three icons for avatars
![Untitled](https://user-images.githubusercontent.com/2261188/54864034-0a2e3180-4d51-11e9-89a1-23612aaf2a08.jpg)

If we use the "white background and grey icon" for multi_user, we'll need to introduce the same for single user avatar. It's an issue, just a thing we should think about before we do it.

Reply:
I'll implement this with what we have at the moment, and if we need to modify it from that we can.

Reply:
I implemented this for now use `AVATAR_WITH_BACKGROUND` and `MULTI_USER_AVATAR_GREY_BACKGROUND`

I couldn't find a way of tinting a vector drawable via XML. So we would need an asset that is tinted the right colour already.

I would like to note that I was also using `AVATAR_WITHOUT_BACKGROUND` for the fallback for the avatar in a `ImageView` that had its own white background.

I agree the grey on blue doesn't look fantastic. Will await what the plan is to resolve that.

Reply:
The icon/avatar issue is resolved in f028444fba60791c0b872cb7a4e8af7a3d2e849b